### ISSUE TITLE
lstm net

### ISSUE THREAD

**Issue by never-to-never** [google/jax#15879]
```
class LSTM(Layer):

  def __init__(self,
               input_size: int,
               hidden_size: int,
               bias: bool = True) -> None:
    super().__init__()
    self.input_size = input_size
    self.hidden_size = hidden_size
    self.w_ii = Linear(in_features = self.input_size,out_features=self.hidden_size, W_initializer=XavierUniformIniter(mode='fan_in', in_axis=1))
    self.w_hi = Linear(in_features = self.hidden_size,out_features=self.hidden_size, W_initializer=XavierUniformIniter(mode='fan_in', in_axis=1))
    self.w_if = Linear(in_features = self.input_size,out_features=self.hidden_size, W_initializer=XavierUniformIniter(mode='fan_in', in_axis=1))
    self.w_hf = Linear(in_features = self.hidden_size,out_features=self.hidden_size, W_initializer=XavierUniformIniter(mode='fan_in', in_axis=1))
    self.w_ig = Linear(in_features = self.input_size,out_features=self.hidden_size, W_initializer=XavierUniformIniter(mode='fan_in', in_axis=1))
    self.w_hg = Linear(in_features = self.hidden_size,out_features=self.hidden_size, W_initializer=XavierUniformIniter(mode='fan_in', in_axis=1))
    self.w_io = Linear(in_features = self.input_size,out_features=self.hidden_size, W_initializer=XavierUniformIniter(mode='fan_in', in_axis=1))
    self.w_ho = Linear(in_features = self.hidden_size,out_features=self.hidden_size, W_initializer=XavierUniformIniter(mode='fan_in', in_axis=1))

    if bias:
      b = nnp.zeros((self.hidden_size * 4,), dtype=float)
      self.b = self.put_variable('b', b)
    c = nnp.zeros((self.hidden_size,), dtype=float)
    self.c = self.put_variable('c', c)

    h = nnp.zeros((self.hidden_size,), dtype=float)
    self.h = self.put_variable('h', h)

  def __call__(self, param, inputs) -> Array:
    self.reset_state(param, inputs.shape[1])
    h = param[self.name]['h']
    c = param[self.name]['c']

    def step(carry, x):
        hx, cx = carry
        hy, cy = self.lstm_cell(param, hx, cx, x)
        return (hy, cy), hy

    _, outputs = jax.lax.scan(step, (h, c), inputs)
    return outputs

  def reset_state(self, param, batch_size):
    c = nnp.zeros((batch_size, self.hidden_size), dtype=float)
    param[self.name]['c'].value = c
    h = nnp.zeros((batch_size, self.hidden_size), dtype=float)
    param[self.name]['h'].value = h

  def lstm_cell(self, param, h, c, input):
    b = param[self.name]['b']
    bi, bf, bg, bo = nnp.split(b, indices_or_sections=4, axis=-1)
    i = nnp.sigmoid(self.w_ii(param, input) + self.w_hi(param, h) + bi)
    f = nnp.sigmoid(self.w_if(param, input) + self.w_hf(param, h) + bf)
    g = nnp.tanh(self.w_ig(param, input) + self.w_hg(param, h) + bg)
    o = nnp.sigmoid(self.w_io(param, input) + self.w_ho(param, h) + bo)
    new_c = f * c + i * g
    new_h = o * nnp.tanh(new_c)
    return new_h, new_c
```
I implemented an LSTM network based on jax. When using this network to train MNIST, the loss of the network basically fluctuates around a fixed value, and the accuracy rate is very low. Please help me to see if there is any problem with my implementation? Thank you so much
Note: The nnp used in the code is my encapsulation of jax.numpy

**Comment by jakevdp** [comment]
Hi - thanks for the question! It's hard to know what's going on without more information. What is `Layer` here? Is it something you defined, or something imported from another library? What data are you fitting the model to? What results are you seeing, vs. what would you expect?

I'd suggest editing your question to add a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the behavior you're seeing.
